internal/runtime/client/client.go

```diff
@@ -117,6 +120,11 @@ func (c *client) Discover(ctx context.Context, extensionConfig *runtimev1.Extens
 		return nil, fmt.Errorf("discovery failed with %v", response.Message)
 	}
 
+	// Check to see if the response is valid.
+	if err = defaultAndValidateDiscoveryResponse(response); err != nil {
```

Comment:
It's seemed like a good idea to bundle this into one function (mostly to save on nil checking), but we can break it out if someone feels strongly about it.

Reply:
I am okay with this approach.